extensions/kamelet/runtime/src/main/java/org/apache/camel/quarkus/component/kamelet/KameletRecorder.java

```diff
@@ -63,6 +63,14 @@ public void configure(CamelContext context) {
                         if (definition.getRoute() != null && definition.getRoute().getOutputs() != null) {
                             definition.getRoute().getOutputs().forEach(o -> o.setCamelContext(context));
                         }
+
+                        if (definition.getTemplateBeans() != null) {
+                            definition.getTemplateBeans()
+                                    .stream()
+                                    .filter(bean -> bean.getResource() != null)
```

Comment:
Is `!=` null really needed as handled by instanceof ?

Reply:
Yes, you're right. I removed the null filter.